library/src/main/java/com/google/android/media/tv/companionlibrary/XmlTvParser.java

```diff
@@ -344,7 +344,7 @@ private static Program parseProgram(XmlPullParser parser)
                 .setChannelId(channelId.hashCode())
                 .setTitle(title)
                 .setDescription(description)
-                .setPosterArtUri(icon.src)
+                .setPosterArtUri(icon != null ? icon.src : null)
```

Comment:
Same as above, since I can't control the the info put by the provider, some channels doesn't have an icon. This could be somehow bypassed by simply creating a "dummy" icon but it doesn't quite solve the problem.